packages/react/src/hooks/__tests__/useAuth.test.ts

```diff
@@ -3,13 +3,31 @@ import { Hub } from '@aws-amplify/core';
 import { act, renderHook } from '@testing-library/react-hooks';
 import { useAuth } from '../useAuth';
 
-jest.mock('@aws-amplify/auth');
+const currentAuthenticatedUser = jest.fn();
+
+jest
+  .spyOn(Auth, 'currentAuthenticatedUser')
+  .mockImplementation(currentAuthenticatedUser);
```

Comment:
Nit: Since we are mocking the resolved value in each test case _think_ this can just be:
```
const currentAuthenticatedUserSpy = jest.spyOn(Auth, 'currentAuthenticatedUser');
```

Reply:
Cleaned up in [9fb0d3f](https://github.com/aws-amplify/amplify-ui/pull/2795/commits/9fb0d3f48700cf6de047b851473069957d343fcf) 🙏